src/ddscxx/include/dds/sub/detail/TDataReaderImpl.hpp

```diff
@@ -479,9 +479,16 @@ dds::sub::detail::DataReader<T>::common_constructor(
             dds::sub::DataReaderListener<T>* listener,
             const dds::core::status::StatusMask& mask)
 {
+#if defined(_MSC_VER)
+#pragma warning( push )
+#pragma warning( disable : 4127 )
+#endif
```

Comment:
I've added a `DDSCXX_MSVC_WARNING_OFF` for this purpose, better use that instead.

Reply:
I will use that then, since I do not like this approach anyway.

Reply:
Seems like this macro is waiting in another PR which is yet to be merged.

Reply:
Should be in `idlcxx` now. Merge the PR last week.

Reply:
Used the DDSCXX macro in stead of the loose pragma statements.